BDArmory/Control/BDModulePilotAI.cs

```diff
@@ -1680,9 +1680,9 @@ void FixedUpdate()
                 if (targetStalenessTimer >= 50) //add some error to the predicted position every second
                 {
                     Vector3 staleTargetPosition = new Vector3();
-                    staleTargetPosition.x = UnityEngine.Random.Range(0f, (float)staleTargetVelocity.magnitude / 2);
-                    staleTargetPosition.y = UnityEngine.Random.Range(0f, (float)staleTargetVelocity.magnitude / 2);
-                    staleTargetPosition.z = UnityEngine.Random.Range((float)staleTargetVelocity.magnitude * 0.75f, (float)staleTargetVelocity.magnitude * 1.25f);
+                    staleTargetPosition.x = UnityEngine.Random.Range(-(float)staleTargetVelocity.magnitude / 2, (float)staleTargetVelocity.magnitude / 2);
```

Comment:
This is adding random noise from a flat distribution. Wouldn't using Gaussian noise from `VectorUtils.GaussianVector3()` be better?
Also, this is in world-space where x,y,z don't corresponding to the local East, North, Up directions, so why are the different axes scaled differently?
Also also, `targetStalenessTime >= 50` is only ticking over every 50s, not every 1s.
Additionally, this is just setting the local copy of `staleTargetPosition`, not the one above `FlyToTargetVessel`, which is always zero and only appears to be being used in
```csharp
target += (staleTargetVelocity * weaponManager.detectedTargetTimeout) + (staleTargetPosition * weaponManager.detectedTargetTimeout);
```

I think you probably want something like this instead:
```csharp
targetStalenessTimer += Time.fixedDeltaTime;
if (targetStalenessTimer >= 1) //add some error to the predicted position every second
{
    staleTargetPosition += staleTargetVelocity.magnitude / 2f * VectorUtils.GaussianVector3();
    targetStalenessTimer = 0;
}
...
target += staleTargetPosition + staleTargetVelocity * weaponManager.detectedTargetTimeout;
```
with `staleTargetPosition` and `staleTargetVelocity` both being reset to zero when there's no longer a stale target.

Reply:
Original concept iirc was to have the AI randomwalk (with new coords every sec) after target went stale as a psuedo-search pattern. Gaussian would be smoother, but with chance for greater deviation from target true position. Either would work, though.

Reply:
Fair enough, I'll revert back to a UnityEngine.Random method.